src/apps/schema/src/app/views/SchemaEdit/FieldSettings/FieldSettings.js

```diff
@@ -77,17 +77,21 @@ export default function FieldSettings(props) {
                   name: formatName(val), // literal name key
                 });
               } else {
-                props.updateValue(val, key);
+                props.updateValue(formatName(val), key);
               }
             }}
           />
           <FieldTypeText
             className={styles.Setting}
             name="name"
             label="Field Name (Parsley Code Reference). Can not contain spaces, uppercase or special characters."
-            defaultValue={props.field.name}
+            {...(props.new
```

Comment:
Where does the `new` property come from?

Reply:
@shrunyan FieldAdd component has FieldSettings child with new prop. FieldEdit component has FieldSettings child without it.

Reply:
Okay. Do we view there is a fundamental issue with how this component is structured and this is a quick workaround or is there something else wrong here and why we need this awkward decision between value or defaultValue?

Reply:
@shrunyan Yea so the cursor jump is a pretty common case within controlled inputs [Check this out](https://stackoverflow.com/questions/28922275/in-reactjs-why-does-setstate-behave-differently-when-called-synchronously/28922465#28922465). The only way around it is either to use defaultValue to avoid initial cursor jump(but this limits the inputs controllability) OR do some much more complex solution regarding cursor management or some async input handler.

In this case we only need to use value as opposed to defaultValue when creating a new field because that is where we are auto-populating the other field and we can stick to using defaultValue on edit and avoid the cursor jump. Hence the conditional use of defaultValue or value depending on whether its a new or an edit. 

IMO we shouldn't worry about this cursor jump because at this point its pretty much expected behavior. So we could just leave it as value for both new and edit.